x-pack/test/plugin_functional/test_suites/global_search/global_search_bar.ts

```diff
@@ -51,22 +52,28 @@ export default function ({ getPageObjects, getService }: FtrProviderContext) {
       it('shows a suggestion when searching for a term matching a type', async () => {
         await navigationalSearch.searchFor('dashboard');
 
-        let results = await navigationalSearch.getDisplayedResults();
-        expect(results[0].label).to.eql('type: dashboard');
-
-        await navigationalSearch.clickOnOption(0);
-        await navigationalSearch.waitForResultsLoaded();
-
-        const searchTerm = await navigationalSearch.getFieldValue();
-        expect(searchTerm).to.eql('type:dashboard');
-
-        results = await navigationalSearch.getDisplayedResults();
-        expect(results.map((result) => result.label)).to.eql([
-          'dashboard 1 (tag-2)',
-          'dashboard 2 (tag-3)',
-          'dashboard 3 (tag-1 and tag-3)',
-          'dashboard 4 (tag-special-chars)',
-        ]);
+        let results;
+        // catching for test failures with label undefined
+        await retry.try(async () => {
+          results = await navigationalSearch.getDisplayedResults();
+          if (results) {
+            expect(results[0].label).to.eql('type: dashboard');
+
+            await navigationalSearch.clickOnOption(0);
+            await navigationalSearch.waitForResultsLoaded();
+
+            const searchTerm = await navigationalSearch.getFieldValue();
+            expect(searchTerm).to.eql('type:dashboard');
+
+            results = await navigationalSearch.getDisplayedResults();
+            expect(results.map((result) => result.label)).to.eql([
+              'dashboard 1 (tag-2)',
+              'dashboard 2 (tag-3)',
+              'dashboard 3 (tag-1 and tag-3)',
+              'dashboard 4 (tag-special-chars)',
+            ]);
+          }
+        });
```

Comment:
Hey! Hey!! I think the retry is a good idea!! My only concern here is that if no display results are found the test assertions would not get called, we could rewrite this like so;

```suggestion
 const dashboardTypeSearchResult = await retry.try(async () => {
          await navigationalSearch.searchFor('dashboard');
          const results = await navigationalSearch.getDisplayedResults();
          // will throw if results not found
          expect(results.length).greaterThan(0);
          return results;
        });

        expect(dashboardTypeSearchResult[0].label).to.eql('type: dashboard');

        await navigationalSearch.clickOnOption(0);
        await navigationalSearch.waitForResultsLoaded();

        const searchTerm = await navigationalSearch.getFieldValue();
        expect(searchTerm).to.eql('type:dashboard');

        const results = await navigationalSearch.getDisplayedResults();
        expect(results.map((result) => result.label)).to.eql([
          'dashboard 1 (tag-2)',
          'dashboard 2 (tag-3)',
          'dashboard 3 (tag-1 and tag-3)',
          'dashboard 4 (tag-special-chars)',
        ]);
```

let me know what you think?